### PR TITLE
`book`: Switch to UUIDs

### PR DESCRIPTION
**Pull request by Urhengulas**
This PR switches the ids to [`uuid::Uuid`](https://docs.rs/uuid/0.8.1/uuid/struct.Uuid.html).
_(See #166 for a different approach)_

* Pro:
    * it seems to be quite adopted in the rust ecosystem
        * `diesel` support it behind `uuidv07` flag
        * `ulid` supports conversion into it behind `uuid` flag
    * it's the same in the database and our model (path, json-response)
* Con:
    * extra layer of abstraction
    * no control over type (e.g. customize `Default` implementation)

### CODE REVIEW

File: book/Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,4 @@ serde = { version = "1.0", features = ["derive"] }
 tarpc = { version = "0.22", features = ["full"] }
 tokio = { version = "0.2", features = ["full"] }
 tokio-serde = { version = "0.6", features = ["json"] }
+uuid = { version = "0.8.1", features = ["serde"] }
```
Comment (code-mm) [inline book/Cargo.toml:27]: Why the patch version?

File: book/migrations/0001_create_base_structure/up.sql [hunk 2]
```diff
@@ -45,29 +45,29 @@ CREATE TABLE books (
...
 CREATE TABLE books_subject_areas (
-    id SERIAL PRIMARY KEY,
-    book_id INTEGER NOT NULL REFERENCES books (id) ON UPDATE CASCADE ON DELETE CASCADE,
-    subject_area_id INTEGER NOT NULL REFERENCES subject_areas (id) ON UPDATE CASCADE ON DELETE CASCADE,
+    id UUID PRIMARY KEY,
+    book_id UUID NOT NULL REFERENCES books ON UPDATE CASCADE ON DELETE CASCADE,
+    subject_area_id UUID NOT NULL REFERENCES subject_areas ON UPDATE CASCADE ON DELETE CASCADE,
     UNIQUE(book_id, subject_area_id)
-);
+);
```
Comment (code-mm) [inline book/migrations/0001_create_base_structure/up.sql:73]: blank line

File: book/migrations/0001_create_base_structure/up.sql [hunk 3]
```diff
@@ -45,29 +45,29 @@ CREATE TABLE books (
     subtitle VARCHAR(500),
     title VARCHAR(100) NOT NULL,
     -- 
-    category_id INTEGER NOT NULL REFERENCES categories (id) ON UPDATE CASCADE ON DELETE RESTRICT,
-    language_id INTEGER NOT NULL REFERENCES languages (id) ON UPDATE CASCADE ON DELETE RESTRICT,
-    publisher_id INTEGER NOT NULL REFERENCES publishers (id) ON UPDATE CASCADE ON DELETE CASCADE
+    category_id INT NOT NULL REFERENCES categories ON UPDATE CASCADE ON DELETE RESTRICT,
+    language_id INT NOT NULL REFERENCES languages ON UPDATE CASCADE ON DELETE RESTRICT,
```
Comment (code-mm) [inline book/migrations/0001_create_base_structure/up.sql:49]: Why no UUID for `category_id` and `language_id`?

**Review by code-mm — CHANGES_REQUESTED** [review]

**Comment by Urhengulas** [comment]
Deferred

**Pull request closed without merging**